app/views/site/_selection_well.html.erb

```diff
@@ -0,0 +1,225 @@
+<% if user_signed_in? %>
+    <script type="text/javascript">
```

Comment:
Is there a reason why this script tag is separate from the lower one?  You're declaring variables here that are referenced lower, so ideally they should be in the same block.